Octokit/Http/HttpClientAdapter.cs

```diff
@@ -42,10 +44,99 @@ public HttpClientAdapter(Func<HttpMessageHandler> getHandler)
+        {
+            var response = await SendRequest(request, cancellationToken).ConfigureAwait(false);
+
+            // Can't redirect without somewhere to redirect too.  Throw?
+            if (response.Headers.Location == null) return response;
+
+            // Don't redirect if we exceed max number of redirects
+            var redirectCount = 0;
+            if (request.Properties.Keys.Contains(RedirectCountKey))
+            {
+                redirectCount = (int)request.Properties[RedirectCountKey];
+            }
+            if (redirectCount > 3)
+            {
+                throw new InvalidOperationException("The redirect count for this request has been exceeded. Aborting.");
+            }
+            request.Properties[RedirectCountKey] = ++redirectCount;
+
+            if (response.StatusCode == HttpStatusCode.MovedPermanently
+                        || response.StatusCode == HttpStatusCode.Redirect
+                        || response.StatusCode == HttpStatusCode.Found
+                        || response.StatusCode == HttpStatusCode.SeeOther
+                        || response.StatusCode == HttpStatusCode.TemporaryRedirect
+                        || (int)response.StatusCode == 308)
+            {
+                var newRequest = CopyRequest(request);
+
+                if (response.StatusCode == HttpStatusCode.SeeOther)
+                {
+                    request.Content = null;
+                    request.Method = HttpMethod.Get;
+                }
+                else
+                {
+                    if (request.Content != null && request.Content.Headers.ContentLength != 0)
+                    {
+                        var stream = await request.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                        if (stream.CanSeek)
+                        {
+                            stream.Position = 0;
+                        }
+                        else
+                        {
+                            throw new Exception("Cannot redirect a request with an unbuffered body");
+                        }
+                        newRequest.Content = new StreamContent(stream);
+                    }
+                    else
+                    {
+                        newRequest.Content = null;
+                    }
+                }
+                newRequest.RequestUri = response.Headers.Location;
+
+                if (string.Compare(newRequest.RequestUri.Host, newRequest.RequestUri.Host, StringComparison.OrdinalIgnoreCase) != 0)
+                {
+                    newRequest.Headers.Authorization = null;
+                }
+
+                response = await GetSuccessfulResponse(newRequest, cancellationToken).ConfigureAwait(false);
+            }
+
+            return response;
+        }
+
+        private async Task<HttpResponseMessage> SendRequest(HttpRequestMessage request, CancellationToken cancellationToken)
```

Comment:
this method isnt really needed as it just `awaits` the call then returns the result without doing anything with it.
I think you could just move the actual `await _http.SendAsync()` call to line #55 where `SendRequest()` is called currently